addons/mail/controllers/thread.py

```diff
@@ -133,6 +133,11 @@ def _prepare_post_data(self, post_data, thread, partner_emails=None, **kwargs):
         if not request.env.user._is_internal():
             partners = partners & self._filter_message_post_partners(thread, partners)
         post_data["partner_ids"] = partners.ids
+        roles = request.env["mail.role"].browse(post_data.pop("role_ids", []))
+        user_ids = request.env["res.users"].search([("mail_role_ids", "in", roles.ids)])
```

Comment:
either use search at first line (to verify existence), or don't use browse and directly pass list of ids to search at 2nd line